internal/resources/connections/data_source_metrics_endpoint_scrape_job.go

```diff
@@ -0,0 +1,111 @@
+package connections
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common/connectionsapi"
+)
+
+type datasourceMetricsEndpointScrapeJob struct {
+	client *connectionsapi.Client
+}
+
+func makeDatasourceMetricsEndpointScrapeJob() *common.DataSource {
+	return common.NewDataSource(
+		common.CategoryConnections,
+		resourceMetricsEndpointScrapeJobTerraformName,
+		&datasourceMetricsEndpointScrapeJob{},
+	)
+}
+
+func (r *datasourceMetricsEndpointScrapeJob) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
```

Comment:
similar comment as earlier for the datasource and resource: most of the pointer receivers can become value receivers. The only exception is this `Configure` function which does modify the datasource/resource's client

Reply:
oh sorry, I deleted a different comment I left on this PR due to your reasoning, and forgot I left this one on here. Thanks for pointing this out